starlark/eval.go

```diff
@@ -46,6 +48,21 @@ type Thread struct {
 	// See example_test.go for some example implementations of Load.
 	Load func(thread *Thread, module string) (StringDict, error)
 
+	// Steps is a count of abstract computation steps executed by
+	// this thread. It is incremented by the interpreter. If after
+	// being incremented, its value is zero, the interpreter calls
+	// thread.Cancel("too many steps").
+	//
+	// Steps may be used as a measure the approximate cost of
+	// Starlark execution, by computing the difference in its value
+	// before and after a computation. It may also be used to limit
+	// a computation to k steps by setting its initial value to
+	// 1<<64 - k.
+	Steps uint64
```

Comment:
No change needed necessarily, just thinking about alternatives:

When I worked on V8, they had a counter which was set to some value, then decremented at function returns and loop back branches. If it reached zero, the profiler was notified, which could trigger optimization.

I think it's a little more intuitive to have a decrementing counter, since you can initially set it to `k` rather than `1<<64 - k`. But then measuring how many steps were taken is harder, so perhaps there's no advantage either way.

In any case, I wonder if it would be better to provide methods to get / set this, rather than exporting the counter itself. There are a lot of uses and possible representations for a counter like this, and it could easily change in the future.